pkg/azure/provider/provider.go

```diff
@@ -98,6 +99,10 @@ func (d defaultDriver) CreateMachine(ctx context.Context, req *driver.CreateMach
 	return
 }
 
+func (d defaultDriver) InitializeMachine(ctx context.Context, request *driver.InitializeMachineRequest) (*driver.InitializeMachineResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "InitializeMachine is not yet implemented by the azure provider")
```

Comment:
should this be an error with `code.Unimplemented` or should it be a no-op. Any call to azure Driver's `InitializeMachine` will not do anything.

Reply:
Well, this can be argued either ways. The reason that I used `codes.Unimplemented` is because that code exists and is actively used in `getVmStatus` and `triggerCreationFlow`. So, I held my nose and stuck to current convention. I prefer code behaviour to be similar to existing code. When we do the refactoring, we can get rid of such non-idiomatic Go practices but for now IMHO it makes sense to follow existing convention.

Reply:
Ok. It just means having unnecessary handling in mcm-core. But since this has been the unfortunate convention we can stick with it till be take up complete refactoring.